xarray/plot/facetgrid.py

```diff
@@ -471,6 +471,73 @@ def add_quiverkey(self, u, v, **kwargs):
         # self._adjust_fig_for_guide(self.quiverkey.text)
         return self
 
+    def _get_largest_lims(self) -> dict[str, tuple[int | float, int | float]]:
+        """
+        Get largest limits in the facetgrid.
+
+        Returns
+        -------
+        lims_largest : dict[str, tuple[int | float, int | float]]
+            Dictionary with the largest limits along each axis.
+
+        Examples
+        --------
+        >>> ds = xr.tutorial.scatter_example_dataset(seed=42)
+        >>> fg = ds.plot.scatter("A", "B", hue="y", row="x", col="w")
+        >>> round(fg._get_largest_lims()["x"][0], 3)
+        -0.334
+        """
+        lims_largest: dict[str, tuple[float, float]] = dict(
+            x=(np.inf, -np.inf), y=(np.inf, -np.inf), z=(np.inf, -np.inf)
+        )
+        for k in ("x", "y", "z"):
+            # Find the plot with the largest xlim values:
+            for ax in self.axes.flat:
+                get_lim: None | Callable[[], tuple[float, float]] = getattr(
+                    ax, f"get_{k}lim", None
+                )
+                if get_lim:
+                    l0, l1 = get_lim()
+                    l0_old, l1_old = lims_largest[k]
+                    lims_largest[k] = (min(l0, l0_old), max(l1, l1_old))
```

Comment:
```suggestion
            l0, l1 = (np.inf, -np.inf)
            for ax in self.axes.flat:
                get_lim: None | Callable[[], tuple[float, float]] = getattr(
                    ax, f"get_{k}lim", None
                )
                if get_lim:
                    l0_new, l1_new = get_lim()
                    l0, l1 = (min(l0, l0_new), max(l1, l1_new))```
            lims_largest[k] = (l0, l1)
```

maybe like this? (untested)

Reply:
But as mentioned - fine to merge as is.

Reply:
Nice one!